components/style/properties/gecko.mako.rs

```diff
@@ -653,6 +654,25 @@ fn static_assert() {
                                corner.y_index,
                                need_clone=True) %>
     % endfor
+
+    pub fn copy_border_image_source_from(&mut self, _other: &Self) {
+        unsafe {
+            Gecko_CopyImageValueFrom(&mut self.gecko.mBorderImageSource,
+                                     &other.gecko.mBorderImageSource);
+        }
+    }
+
+    pub fn set_border_image_source(&mut self, v: longhands::border_image_source::computed_value::T) {
+        unsafe {
+            // Prevent leaking of the last elements we did set
+            Gecko_SetNullImageValue(&mut self.gecko.mBorderImageSource);
+        }
+
+        if let Some(image) = v.0 {
+            set_image(image, &mut self.gecko.mBorderImageSource)
```

Comment:
fwiw this could be a method on `nsStyleImage` in gecko_conversions.rs. But not necessary, this is fine too.